code/game/machinery/doors/firedoor.dm

```diff
@@ -203,30 +199,90 @@
 		for(var/obj/machinery/firealarm/fire_panel in place.firealarms)
 			fire_panel.set_status()
 
-/obj/machinery/door/firedoor/proc/check_atmos(datum/source)
+/obj/machinery/door/firedoor/proc/merger_adding(obj/machinery/door/firedoor/us, datum/merger/new_merger)
+	SIGNAL_HANDLER
+	if(new_merger.id != merger_id)
+		return
+	RegisterSignal(new_merger, COMSIG_MERGER_REFRESH_COMPLETE, .proc/refresh_shared_turfs)
+
+/obj/machinery/door/firedoor/proc/merger_removing(obj/machinery/door/firedoor/us, datum/merger/old_merger)
+	SIGNAL_HANDLER
+	if(old_merger.id != merger_id)
+		return
+	UnregisterSignal(old_merger, COMSIG_MERGER_REFRESH_COMPLETE)
+
+/obj/machinery/door/firedoor/proc/refresh_shared_turfs(datum/source, list/leaving_members, list/joining_members)
+	SIGNAL_HANDLER
+	var/datum/merger/temp_group = source
+	var/datum/merger/merge_group = GetMergeGroup(merger_id, merger_typecache)
+	if(temp_group.origin != src)
+		return
+	var/list/shared_problems = list() // We only want to do this once, this is a nice way of pulling that off
+	for(var/obj/machinery/door/firedoor/firelock as anything in merge_group.members)
+		firelock.issue_turfs = shared_problems
+
+/obj/machinery/door/firedoor/proc/register_adjacent_turfs(atom/loc)
+	for(var/dir in GLOB.cardinals)
+		var/turf/checked_turf = get_step(get_turf(loc),dir)
+		process_results(checked_turf)
+		if(checked_turf)
+			RegisterSignal(checked_turf, COMSIG_TURF_EXPOSE, .proc/process_results)
+			RegisterSignal(checked_turf, COMSIG_TURF_CALCULATED_ADJACENT_ATMOS, .proc/process_results)
+
+
+/obj/machinery/door/firedoor/proc/unregister_adjacent_turfs(atom/loc)
+	for(var/dir in GLOB.cardinals)
+		var/turf/checked_turf = get_step(get_turf(loc),dir)
+		process_results(checked_turf)
+		if(checked_turf)
+			UnregisterSignal(checked_turf, COMSIG_TURF_EXPOSE)
+			UnregisterSignal(checked_turf, COMSIG_TURF_CALCULATED_ADJACENT_ATMOS)
+
+/obj/machinery/door/firedoor/proc/check_atmos(turf/checked_turf)
+	var/datum/gas_mixture/environment = checked_turf.return_air()
+	var/obj/machinery/airalarm/air_alarm = LAZYACCESS(my_area.airalarms,1)
```

Comment:
not sure how I feel about this

Reply:
why not?

Reply:
Cause the settings are not the same across all air alarms in an area is the thing, so it'll be really unintuitive for people to change, which makes me question why it exists

Reply:
Is this meant to be something people actually do?

Reply:
I mean they could perhaps

Reply:
I'd like it removed for now I think. I don't want to add something this out of the way, would prefer something more fully implemented

Reply:
done